### PR TITLE
Perforce/19.1

### PR DESCRIPTION
**Pull request by danielhompanera**
### Purpose of this PR
The native perforce plugin does not support MFA (Multi Factor Authentication). This was introduced in v18.2 of the Perforce API (currrently plugin points to v17.2). Therefore, as a first step we need to update perforce update to the selected version 19.1

---
### Testing status

**Manual Tests**: 
- [X] Verified that plugins work in the Editor
- [X] Ran local tests on your machine

**Automated Tests**: 
- Perforce tests updated and passing in Yamato

**Yamato**: (Select your branch):
https://unity-ci.cds.internal.unity3d.com/project/421/branch/perforce%2F19.1

---
### Comments to reviewers
We started this upgrade from an old branch perforce/mfa that was updating Perforce API to 18.2. (that branch can be removed as well once this PR is merged).
Talking to the owner of the branch @tomaskucinskas they found some problems when pushing it to yamato. This was probably due to the fact that they updated VS build tools to 2015 from 2010. 
In this branch we update Perforce API to 19.1 and take the build tools back to 2010. The main change in v19.1 is the introduction of 4 new dependencies (libp4script libp4script_c libp4script_curl libp4script_sqlite) that have been added as external libraries for the three platforms. Special mention to Linux where we needed to add a GTK3 include for harfbuzz to make it locally buildable.

### CODE REVIEW

File: Test/Source/TestServer.cpp [hunk 1]
```diff
@@ -438,8 +438,9 @@ static int runScript(ExternalProcess& p, const std::string& testDir, const std::
 	}
 	if (lineNum == 0)
 	{
+		std::cerr << "Invalid test script it has no lines " << scriptPath.c_str() << std::endl;
 		//Won't compile with gcc >= 6.1
-		std::cerr << "Invalid test script it has no lines " << testscript << std::endl;
+		//std::cerr << "Invalid test script it has no lines " << testscript << std::endl;
```
Comment (mig42) [inline Test/Source/TestServer.cpp:443]: Should we remove this line entirely?

File: P4Plugin/Source/P4LoginCommand.cpp [hunk 2]
```diff
@@ -19,6 +19,7 @@ class P4LoginCommand : public P4Command
 
 		m_LoggedIn = false;
 		m_Password = task.GetP4Password();
+		//m_CheckingForLoggedIn = args.size() > 1;
```
Comment (mig42) [inline P4Plugin/Source/P4LoginCommand.cpp:22]: Can this be removed? What's the consequence for not setting `m_CheckingForLoggedIn`?

**Review by mig42 — CHANGES_REQUESTED** [review]
There are a couple of comments that maybe we can get rid of, and also a question about one of them 🙂  Nice work though! Upgrading the API version is never an easy task, this is great 😄

File: P4Plugin/Source/P4LoginCommand.cpp (hunk 2 above)
Comment (danielhompanera) [reply P4Plugin/Source/P4LoginCommand.cpp:22]: This is a waste from the rebase from master we did. Removing theme

File: Test/Source/TestServer.cpp (hunk 1 above)
Comment (danielhompanera) [reply Test/Source/TestServer.cpp:443]: Removing...

**Review by mig42 — APPROVED** [review]
Looks Great To Me

**Review by AscendingMan — APPROVED** [review]

**Pull request merged**